project/src/main/java/com/google/sps/Models/Database.java

```diff
@@ -99,6 +102,85 @@ private static User createUser(String email, String nickname) {
     getDatastore().put(userEntity);
     long userID = userEntity.getKey().getId();
 
-    return new User(email, nickname, userID);
+    return new User(email, nickname, userID, documents);
+  }
+
+  private static void addDocumentForUser(String hash, long userID) {
+    Query query = new Query("User").addFilter("__key__", Query.FilterOperator.EQUAL, KeyFactory.createKey("User", userID));
+    Entity userEntity = getDatastore().prepare(query).asSingleEntity();
+    ArrayList<String> docHashes = getUsersDocumentsHashes(userID);
+
+    docHashes.add(hash);
+    //user.setDocs(docHashes);
+    userEntity.setProperty("documents", docHashes);
+  }
+
+  public static Document createDocument(String name, String language, String hash, long userID) {
+      // I believe a static version would suit our needs better
+      // as when you share it with someone their ID gets appended to the array.
+      Entity docEntity = new Entity("Document");
+      ArrayList<Long> userIDs = new ArrayList<Long>();
+      
+      docEntity.setProperty("name", name);
+      docEntity.setProperty("language", language);
+      docEntity.setProperty("hash", hash);
+      userIDs.add(userID);
+      docEntity.setProperty("userIDs", userIDs);
+      getDatastore().put(docEntity);
+
+      return new Document(name, language, hash, userIDs);
+  }
+
+  public static Document getDocumentByHash(String hash) {
+    Query query = new Query("Document").addFilter("hash", Query.FilterOperator.EQUAL, hash);
+    Entity docEntity = getDatastore().prepare(query).asSingleEntity();
+
+    String name = (String) docEntity.getProperty("name");
+    String language = (String) docEntity.getProperty("language");
+    ArrayList<Long> userIDs = (ArrayList)docEntity.getProperty("userIDs");
+
+    if(docEntity == null) {
+      return null;
+    }
+
+    return new Document(name, language, hash, userIDs);
+  }
+
+  public static ArrayList<Long> getDocumentUsers(String hash) {
```

Comment:
Is this function used anywhere? If not, add it when it's used.

Reply:
Sounds good!